hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bloom/IntervalTreeBasedGlobalIndexFileFilter.java

```diff
@@ -51,10 +50,6 @@ class IntervalTreeBasedGlobalIndexFileFilter implements IndexFileFilter {
       allIndexFiles.add(file);
     }));
 
-    // Note that the interval tree implementation doesn't have auto-balancing to ensure logN search time.
-    // So, we are shuffling the input here hoping the tree will not have any skewness. If not, the tree could be skewed
-    // which could result in N search time instead of NlogN.
-    Collections.shuffle(allIndexFiles);
```

Comment:
@yabola do you have some micro-benchmark as to how much improvement this change brings?

Reply:
Sorry, actually I don't have benchmark on it. I think red-black tree is a general optimization strategy like hashmap